theme/layouts/_sidebar.ejs

```diff
@@ -0,0 +1,35 @@
+<%
+const query = site.queryPages;
+subPages = site.getSubPages;
+
+const navigation = [];
+for (const menuItem of site.menuItems) {
+  const page = query(({originalPathname}) => originalPathname === menuItem)[0];
+  navigation.push(subPages(page));
+}
+
+const [currentPage] = query(({pathname}) => page.pathname === pathname);
+let currentPathname = currentPage.originalPathname.split("/");
+pathnameList = [];
+for (let i=currentPathname.length-1; i >= 0; i--) {
+  pathnameList.push(currentPathname.join("/"));
+  currentPathname = currentPathname.slice(0,i);
+}
+
+  function recNav(navig) { %>
+    <ul>
+      <% for (const {title, pathname, originalPathname, sub_items} of navig) { %>
+        <li <%- originalPathname.split("/").length == 1 ? "class='primary-menu'" : "" %>>
```

Comment:
We don't need to specify `.primary-menu` class to style top most list items to do that you can just use direct children selector `#sidebar ul > li`.